controllers/routemonitor/deleter/deleter.go

```diff
@@ -52,3 +52,25 @@ func (r *RouteMonitorDeleter) EnsureServiceMonitorResourceAbsent(ctx context.Con
 	}
 	return nil
 }
+
+func (r *RouteMonitorDeleter) EnsurePrometheusRuleResourceAbsent(ctx context.Context, routeMonitor v1alpha1.RouteMonitor) error {
+	namespacedName := types.NamespacedName{Name: routeMonitor.Status.PrometheusRuleRef.Name,
```

Comment:
what happens if name and namespace are empty because no availabilityTarget is specified? Shouldn't that do an early exit?

Reply:
yes, that is a valid concern, will add them